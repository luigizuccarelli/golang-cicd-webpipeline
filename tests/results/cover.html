
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>handlers: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go (91.8%)</option>
				
				<option value="file1">lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package handlers

import (
        "bytes"
        "fmt"
        "html/template"
        "io/ioutil"
        "net/http"
        "os"
        "sort"
        "strings"
        "time"

        "github.com/gorilla/mux"
        "lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/connectors"
        "lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/schema"
)

const (
        CONTENTTYPE     string = "Content-Type"
        APPLICATIONJSON string = "application/json"
)

func ResultsHandler(w http.ResponseWriter, r *http.Request, conn connectors.Clients) <span class="cov8" title="1">{
        var res *schema.ResultsSchema

        vars := mux.Vars(r)

        if vars["project"] == "" </span><span class="cov8" title="1">{
                w.WriteHeader(http.StatusInternalServerError)
                fmt.Fprintf(w, "%s", "ERROR missing param")
                return
        }</span>

        // open the directory for the project
        <span class="cov8" title="1">files, err := ioutil.ReadDir(os.Getenv("CICD_CONSOLE_DIR") + "/" + vars["project"])
        if err != nil </span><span class="cov8" title="1">{
                w.WriteHeader(http.StatusInternalServerError)
                fmt.Fprintf(w, "%s", "ERROR project folder")
                return
        }</span>

        <span class="cov8" title="1">sort.Slice(files, func(i, j int) bool </span><span class="cov8" title="1">{
                return files[i].ModTime().Before(files[j].ModTime())
        }</span>)

        <span class="cov8" title="1">fmt.Println("DEBUG LMZ", files)

        var items []schema.ItemInfo
        var item = &amp;schema.ItemInfo{}
        for _, f := range files </span><span class="cov8" title="1">{
                // ignore errors from pipelinecontent
                data, _ := ioutil.ReadFile(os.Getenv("CICD_CONSOLE_DIR") + "/" + vars["project"] + "/" + f.Name())
                item.Name = strings.ToUpper(f.Name()[:len(f.Name())-4])
                item.Time = f.ModTime().Unix()
                unixTimeUTC := time.Unix(f.ModTime().Unix(), 0)
                item.DisplayTime = fmt.Sprintf("%v", unixTimeUTC)
                contents := string(data)
                item.Log = contents
                if contents[0:3] == "PAS" </span><span class="cov8" title="1">{
                        item.Pass = true
                        item.Fail = false
                }</span>
                <span class="cov8" title="1">if contents[0:3] == "ERR" </span><span class="cov8" title="1">{
                        item.Pass = false
                        item.Fail = true
                }</span>
                <span class="cov8" title="1">items = append(items, *item)
                item = &amp;schema.ItemInfo{}</span>
        }
        <span class="cov8" title="1">addHeaders(w,r)
        res = &amp;schema.ResultsSchema{StartTime: items[0].DisplayTime, StopTime: items[len(items)-1].DisplayTime, Items: items}
        html, err := ioutil.ReadFile(os.Getenv("TEMPLATE_FILE"))
        if err != nil </span><span class="cov8" title="1">{
                conn.Error("Rsultshandler %v", err)
                w.WriteHeader(http.StatusInternalServerError)
                fmt.Fprintf(w, "%s", "ERROR creating template")
                return
        }</span>
        <span class="cov8" title="1">var tpl bytes.Buffer
        tmpl := template.New(vars["project"])
        tmp, er := tmpl.Parse(string(html))
        if er != nil </span><span class="cov0" title="0">{
                conn.Error("Rsultshandler %v", er)
                w.WriteHeader(http.StatusInternalServerError)
                fmt.Fprintf(w, "%s", "ERROR parsing template")
                return
        }</span>
        <span class="cov8" title="1">tmp.Execute(&amp;tpl, res)
        conn.Debug("ResultsHandler response : %s", tpl.String())
        w.WriteHeader(http.StatusOK)
        fmt.Fprintf(w, "%s", tpl.String())</span>
}

func IsAlive(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        addHeaders(w, r)
        fmt.Fprintf(w, "{ \"version\" : \""+os.Getenv("VERSION")+"\" , \"name\": \""+os.Getenv("NAME")+"\" }")
        return
}</span>

// headers (with cors) utility
func addHeaders(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        if r.Header.Get("API-KEY") != "" </span><span class="cov0" title="0">{
                w.Header().Set("API_KEY_PT", r.Header.Get("API_KEY"))
        }</span>
        <span class="cov8" title="1">w.Header().Set(CONTENTTYPE, APPLICATIONJSON)
        // use this for cors
        w.Header().Set("Access-Control-Allow-Origin", "*")
        w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
        w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization, Accept-Language")</span>
}
</pre>
		
		<pre class="file" id="file1" style="display: none">package validator

import (
        "errors"
        "fmt"
        "os"
        "strconv"
        "strings"

        "github.com/microlib/logger/pkg/multi"
)

// checkEnvars - private function, iterates through each item and checks the required field
func checkEnvar(item string, logger *multi.Logger) error <span class="cov8" title="1">{
        name := strings.Split(item, ",")[0]
        required, _ := strconv.ParseBool(strings.Split(item, ",")[1])
        logger.Trace(fmt.Sprintf("Input paramaters -&gt; name %s : required %t", name, required))
        if os.Getenv(name) == "" </span><span class="cov8" title="1">{
                if required </span><span class="cov8" title="1">{
                        logger.Errorf("%s envar is mandatory please set it", name)
                        return errors.New(fmt.Sprintf("%s envar is mandatory please set it", name))
                }</span>

                <span class="cov8" title="1">logger.Errorf(fmt.Sprintf("%s envar is empty please set it", name))</span>
        }
        <span class="cov8" title="1">return nil</span>
}

// ValidateEnvars : public call that groups all envar validations
// These envars are set via the openshift template
func ValidateEnvars(logger *multi.Logger) error <span class="cov8" title="1">{
        items := []string{
                "LOG_LEVEL,false",
                "NAME,false",
                "SERVER_PORT,true",
                "VERSION,true",
                "TEMPLATE_FILE,true",
                "CICD_CONSOLE_DIR,true",
        }
        for x := range items </span><span class="cov8" title="1">{
                if err := checkEnvar(items[x], logger); err != nil </span><span class="cov8" title="1">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
