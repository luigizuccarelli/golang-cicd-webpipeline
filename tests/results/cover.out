mode: set
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:13.58,17.27 4 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:25.2,25.12 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:17.27,18.15 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:23.3,23.70 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:18.15,21.4 2 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:30.49,39.23 2 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:44.2,44.12 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:39.23,40.54 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/validator/validate.go:40.54,42.4 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:24.86,29.27 3 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:36.2,37.16 2 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:43.2,43.40 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:49.2,51.26 3 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:71.2,74.16 4 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:80.2,83.15 4 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:89.2,90.16 2 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:93.2,95.36 3 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:29.27,33.3 3 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:37.16,41.3 3 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:43.40,45.3 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:51.26,60.29 8 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:64.3,64.29 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:68.3,69.28 2 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:60.29,63.4 2 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:64.29,67.4 2 0
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:74.16,79.3 4 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:83.15,88.3 4 0
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:90.16,92.3 1 0
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:98.54,101.2 2 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:104.57,105.35 1 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:108.2,112.151 4 1
lmzsoftware.com/lzuccarelli/golang-cicd-webconsole/pkg/handlers/handler.go:105.35,107.3 1 0
